jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnectionFactory.java

```diff
@@ -58,18 +57,27 @@ public HttpConfiguration getHttpConfiguration()
     @Override
     public void configure(Connector connector)
     {
-        if (isRecordHttpComplianceViolations())
+        // TODO: need HTTP/2 and HTTP/3 version of this
+        if (getHttpConfiguration().isNotifyComplianceViolations())
```

Comment:
Yeah this is screaming too me now to just add the listeners to the HttpConfiguration. Why look at the configuration to decide if to add a bean that is looked for elsewhere.